### PR TITLE
Use GitHub Actions for CI

### PR DESCRIPTION
**Pull request by gfx**
This is simply porting .travis.yml to workfows/ci.yml. Nothing has been added nor removed, I believe.

### CODE REVIEW

File: misc/docker-ci/check.mk
Comment (kazuho) [inline misc/docker-ci/check.mk:6]: I think the only concerns regarding this pull request are in the changes to this file.
* Re removing `-it`, I agree that we do not need to open TTY, and I think it's correct to drop them. If anything bad happens, we can revive them.
* Re removing `-e TRAVIS`, discussed offline with @gfx.

**Review by kazuho — COMMENTED** [review]
Thank you for the PR. It is my understanding that this PR is still work-in-progress, but it looks great. I'm happy to see how cleanly our CI process is ported to GitHub Actions.

File: misc/docker-ci/check.mk
Comment (gfx) [reply misc/docker-ci/check.mk:6]: Re `-it`: It seems that GitHub Actions have different configurations from Travis about TTY, so `docker run -it` fails with the error "the input device is not a TTY", so we should find workarounds if it gets required.

Comment (kazuho) [reply misc/docker-ci/check.mk:6]: Thank you for the clarification. If some environments happen to require the use of `-it`, we can always pass that as an argument to make running this makefile.

**Comment by gfx** [comment]
https://github.com/h2o/h2o/pull/2501/checks?check_run_id=1496032071

```
            #   Failed test at t/40tls13-early-data.t line 81.
            #                   'd�/�v��G`K������� J0�a��a��J_)�~�n��F�
7count:2'
            #     doesn't match '(?^si:\x01\x04\x00\x00\x00\x01\x88.*\x00[\x00\x01]\x00\x00\x00\x01count:3)'
            # Looks like you failed 1 test of 2.
        
        #   Failed test 'sleep=0'
        #   at t/40tls13-early-data.t line 118.
Plack::Handler::Starlet: Accepting connections at http://127.0.0.1:45707/
        # Looks like you failed 1 test of 2.
    
    #   Failed test 'proxy'
    #   at t/40tls13-early-data.t line 120.
FastCGI: manager (pid 16330): initialized
FastCGI: manager (pid 16330): server (pid 16331) started
FastCGI: server (pid 16331): initialized
FastCGI: manager (pid 16330): received signal TERM
FastCGI: manager (pid 16330): sending TERM to PIDs, 16331
FastCGI: manager (pid 16330): server (pid 16331) exited with status 0
FastCGI: manager (pid 16330): dying: safe exit from signal TERM
FastCGI: manager (pid 16334): initialized
FastCGI: manager (pid 16334): server (pid 16335) started
FastCGI: server (pid 16335): initialized
FastCGI: manager (pid 16334): received signal TERM
FastCGI: manager (pid 16334): sending TERM to PIDs, 16335
FastCGI: manager (pid 16334): server (pid 16335) exited with status 0
FastCGI: manager (pid 16334): dying: safe exit from signal TERM
killing /home/ci/build/h2o... received SIGTERM, gracefully shutting down
killed (got 0)
    # Looks like you failed 1 test of 3.

#   Failed test 'http/2'
#   at t/40tls13-early-data.t line 83.
# Looks like you failed 1 test of 2.
t/40tls13-early-data.t ............................... 
    # Subtest: http/1
        # Subtest: proxy
            # Subtest: sleep=0
            ok 1
            ok 2
            ok 3
            1..3
        ok 1 - sleep=0
            # Subtest: sleep=1
            ok 1
            ok 2
            ok 3
            1..3
        ok 2 - sleep=1
        1..2
    ok 1 - proxy
        # Subtest: fcgi
            # Subtest: sleep=0
            ok 1
            ok 2
            ok 3
            1..3
        ok 1 - sleep=0
            # Subtest: sleep=1
            ok 1
            ok 2
            ok 3
            1..3
        ok 2 - sleep=1
        1..2
    ok 2 - fcgi
        # Subtest: mruby
        ok 1
        ok 2
        ok 3
        1..3
    ok 3 - mruby
    1..3
ok 1 - http/1
    # Subtest: http/2
        # Subtest: proxy
            # Subtest: sleep=0
            ok 1
            not ok 2
            1..2
        not ok 1 - sleep=0
            # Subtest: sleep=1
            ok 1
            ok 2
            1..2
        ok 2 - sleep=1
        1..2
    not ok 1 - proxy
        # Subtest: fcgi
            # Subtest: sleep=0
            ok 1
            ok 2
            1..2
        ok 1 - sleep=0
            # Subtest: sleep=1
            ok 1
            ok 2
            1..2
        ok 2 - sleep=1
        1..2
    ok 2 - fcgi
        # Subtest: mruby
        ok 1
        ok 2
        1..2
    ok 3 - mruby
    1..3
not ok 2 - http/2
1..2
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/2 subtests 
```

**Comment by kazuho** [comment]
FWIW I tend to prefer d295acf. The rationale is that stdout and stderr do not coming out in the correct order is a problem of GitHub Actions. Then, the fix should go into the file specific to GitHub Actions.

**Review by kazuho — APPROVED** [review]
LGTM.

@gfx and I have agreed that:
* This PR will be purely additional in sense that it adds GitHub Actions as the CI runner.
  * The only exception is the removal of the `TRAVIS` environment variable. The assumption is that downstreams can make necessary adjustments.
* When running tests on GitHub Actions, we still see frequent failure in t/40tls13-early-data.t. @kazuho will look into it.
* We might stop using Travis CI, but that would happen only after we gain enough experience running CI using GitHub Actions successfully.
* We will continue allowing CI failures. Reviewers should review the code _and_ CI status before merging the pull requests.

**Pull request merged**
